app/Database/Migrations.hs

```diff
@@ -0,0 +1,37 @@
+module Database.Migrations
+    (migrateDatabase) where
+
+import Control.Monad.Reader (MonadIO)
+import Database.Database (getDatabaseVersion)
+import Database.Persist.Sql (Migration, SqlPersistT, addMigration, runMigration, runMigrationUnsafe)
+import Database.Tables (migrateAll)
+
+data MigrationWrapper = MigrationWrapper {
+    version :: Int,
+    script :: Migration
+}
+
+-- | Migrates the database
+migrateDatabase :: MonadIO m => SqlPersistT m ()
+migrateDatabase = do
+    -- Run unsafe migrations
+    currVersion <- getDatabaseVersion
+    applyMigrations currVersion migrationList
+    -- Run safe migrations
+    runMigration migrateAll
+
+-- | Migrates the database by applying only migrations newer than the current version number
+applyMigrations :: MonadIO m => Int -> [MigrationWrapper] -> SqlPersistT m ()
+applyMigrations currVersion migrations = do
+    mapM_ (runMigrationUnsafe . script) $ filter (\migration -> version migration > currVersion) migrations
```

Comment:
let's sort by version number as well (i.e., don't assume `migrations` is already sorted)